src/replace/imitate.coffee

```diff
@@ -10,9 +10,18 @@ wrapWithConstructor = require('./wrap-with-constructor')
 stringifyAnything = require('../stringify/anything')
 
 module.exports = (realThing, optionalName) ->
+  if canImitate(realThing)
+    imitate(realThing, optionalName)
+  else
+    log.error("td.replace", "\"#{optionalName}\" property was found, but test double only knows how to replace functions, constructors, & objects containing functions (its value was #{stringifyAnything(realThing)}).")
+
+module.exports.canImitate = canImitate
```

Comment:
The other option here is to accept an arg that would skip erroring on demand, but logic would still be required to avoid adding empty imitations to the fakeThing.